setuptools/dist.py

```diff
@@ -361,6 +357,26 @@ def __init__(self, attrs=None):
                 )
         self._finalize_requires()
 
+    def _install_setup_requires(self, attrs):
+        self.dependency_links = attrs.get('dependency_links')
```

Comment:
did you drop [] fallback value intentionally?

Reply:
Yes, using `setup(dependency_links=[])` should override a corresponding `dependency_links` setting in the configuration. If not passed as argument to `setup()` or set in the configuration, [the default value will still be set](https://github.com/pypa/setuptools/pull/1150/files/6fa3dd352774102e567a03b998a57175e71178d2#diff-0f33d7d69d9f143736a252a75373261bR375).